src/main/kotlin/org/kafkalytic/plugin/KafkaUtils.kt

```diff
@@ -254,15 +267,36 @@ fun produceGeneratedMessages(producer: KafkaProducer<ByteArray, ByteArray>, topi
     notify("Produced total $futures messages for topic $topic")
 }
 
-fun produceSingleMessage(producer: KafkaProducer<ByteArray, ByteArray>, topic: String, key: String, value: ByteArray) {
+fun produceSingleMessage(producer: KafkaProducer<ByteArray, ByteArray>, topic: String, key: String, value: ByteArray, header: String) {
     try {
-        producer.send(ProducerRecord<ByteArray, ByteArray>(topic, key.toByteArray(), value)).get()
+        val message = ProducerRecord<ByteArray, ByteArray>(topic, key.toByteArray(), value)
+        if(!header.isNullOrEmpty()){
+            //Custom WITHOUT HEADER
+            val header = createCustomHeader(header);
+            header.forEach {
+                message.headers().add(it);
+            }
+        }
```

Comment:
val message = ProducerRecord(topic, null, key.toByteArray(), value, header?.let { createCustomHeader(it) })